File: 7.Reverse Integer/JavaScript/Solution-2016-01-13.js
/**
 * @param {number} x
 * @return {number}
 */
var reverse = function(x) {
    var flag = 1;
    var y = Math.abs(x);
    var sum = 0;
    var temp;
    var max = 2147483647;
    
    if(x === 0) {
        return 0;
    }
    else if(x < 0) {
        flag = -1;
    }
    
    while(y) {
        temp = y % 10;
        if(sum > parseInt((max - temp) / 10)) {
            return 0;
        }
        sum = sum * 10 + temp;
        y = parseInt(y / 10);
    }
    
    return flag * sum;
};

File: 7.Reverse Integer/C/Solution-2016-01-13.cpp
int reverse(int x) {
    int flag=1;
    int sum=0;
    int temp;
    int y=abs(x);
    
    if(x==0) {
        return 0;
    }
    else if(x<0) {
        flag=-1;
    }
    
    while(y) {
        temp=y%10;
        if(sum>(INT_MAX-temp)/10) {
            return 0;
        }
        sum=sum*10+temp;
        y=y/10;
    }
    
    return flag*sum;
}